studio/src/mas-hash-manager.js

```diff
@@ -0,0 +1,62 @@
+import { LitElement, nothing } from 'lit';
+import StoreController from './reactivity/store-controller.js';
+import Store from './store.js';
+import { setHashParams } from './utils.js';
+import MasSearch from './entities/search.js';
+import MasFilters from './entities/filters.js';
+
+class MasHashManager extends LitElement {
```

Comment:
why this need to be a web component?
why deeplink.js not sufficient?

Reply:
Hm, actually looking over it, it doesn't have to be a component. It can just be a simple function.
Deeplink does what store subscription does, so no point to have both and make things confusing.

In a simple function we can subscribe to the `search` and `filters` stores, and to the page hash (`hashchange`), and make them update each other.

I would prefer to do this separately, after merge if that's ok.

Reply:
To update the state, we can either update hash, react to hash changes and update the store with these values
or update the store and the hash but not react to hash changes, after the first initialization.
both are possible with deeplink.js.

Reply:
We can have something like:

![image](https://github.com/user-attachments/assets/2e331c67-49d2-4999-aec7-a2deaa52704e)
And make sure they don't enter an infinite update loop.

Maybe we would even want to merge the search and filters stores, to be honest.

My point is, we don't really need the `deeplink.js` anymore, which is supposed to be used in `connectedCallback` and has a disposer, when we can just call a function once that subscribes to stuff, and that's it. The state is outside the components now.

Reply:
agree it could be a function, but yeah, could be done in next PR